=====================
Evaluate predictions
=====================


**See the scripts used for downstream evaluation** `here`_ **.**

| 1. Run `BUSCO` to see how well the conserved genes are represented in your final predictions

| 2. Run `OrthoFinder` to find and annotate orthologs present in your predictions

| 3. Run `phylostratR` to find orphan genes in your predictions

| 4. Add functional annotation to your genes using homology and `InterProScan`


.. _here: https://github.com/eswlab/orphan-prediction/tree/master/scripts/downstream
