BRAKER prediction
==================

**See the scripts used for BRAKER** `here`_ **.**

- Input files for Braker

  .. code-block:: yaml

      RNA-Seq raw reads fastq files: *_1.fastq.gz, *_2.fastq.gz
      reference genome fasta file: TAIR10_chr_all.fas


- To simplify handling of files, combine all the forward reads to one file and all the reverse reads to another.

   .. code-block:: bash
       :linenos:

       cat *_1.fastq.gz >> forward_reads.fq.gz
       cat *_2.fastq.gz >> reverse_reads.fq.gz


- Do RNA-Seq alignment by Hisat2:

   .. code-block:: bash
       :linenos:

       ./01_runHisat2.sh forward_reads.fq.gz reverse_reads.fq.gz TAIR10_chr_all.fas


- Run BRAKER by your installed tools:

   .. code-block:: bash
       :linenos:

       ./02_runBraker.sh TAIR10_chr_all.fas TAIR10_rnaseq.bam


- Run BRAKER by singularity comtainer:

  **If you have difficulty to install BRAKER, we suggest you to download the** `singularity container`_ **for this step.**

  To use the container, follow the instruction of the container to install and copy directory.

  .. code-block:: bash
      :linenos:

      ./02_runBraker_singularity.sh TAIR10_chr_all.fas TAIR10_rnaseq.bam


*Note: We only used RNA-Seq as evidence in our cases. BRAKER also accept protein evidence in different prediction mode. However, It is not always best to use all evidence. You can test and determine which one is best for the species you are working on. See the difference among BRAKER modes on their* `main page`_.


.. _here: https://github.com/eswlab/orphan-prediction/tree/master/scripts/braker
.. _main page: https://github.com/Gaius-Augustus/BRAKER#running-braker
.. _singularity container: https://github.com/aseetharam/braker
